openfe/protocols/openmm_rfe/_rfe_utils/relative.py

```diff
@@ -2346,21 +2347,99 @@ def _create_mdtraj_topology(self):
             at1_uniq = at1_hybrid_idx in self._atom_classes['unique_new_atoms']
             at2_uniq = at2_hybrid_idx in self._atom_classes['unique_new_atoms']
             if at1_uniq or at2_uniq:
-                if atom1.index in self._atom_classes['unique_new_atoms']:
-                    atom1_to_bond = added_atoms[atom1.index]
+                if at1_uniq:
+                    atom1_to_bond = added_atoms[at1_hybrid_idx]
                 else:
-                    atom1_to_bond = atom1
+                    old_idx = self._hybrid_to_old_map[at1_hybrid_idx]
+                    atom1_to_bond = hybrid_topology.atom(old_idx)
 
-                if atom2.index in self._atom_classes['unique_new_atoms']:
-                    atom2_to_bond = added_atoms[atom2.index]
+                if at2_uniq:
+                    atom2_to_bond = added_atoms[at2_hybrid_idx]
                 else:
-                    atom2_to_bond = atom2
+                    old_idx = self._hybrid_to_old_map[at2_hybrid_idx]
+                    atom2_to_bond = hybrid_topology.atom(old_idx)
 
                 hybrid_topology.add_bond(atom1_to_bond, atom2_to_bond)
 
         return hybrid_topology
 
 
+    def _create_hybrid_topology(self):
```

Comment:
So we can't do openmm -> MDTraj -> openmm because indices will get renumered.

Instead here we just directly create a hybrid Topology from the input old / new topologies

Reply:
This doesn't have as much utility now, but the _old_ roundtrip method is still incorrect - at least this one doesn't renumber indices (although the residue will be different than those created by the MDTraj trajectory).

I would like to keep this around so that we can switch to this in 1.x?